src/v/cloud_storage/remote.cc

```diff
@@ -218,10 +218,17 @@ ss::future<download_result> remote::download_manifest(
           bucket,
           path);
         result = download_result::timedout;
-    } else {
```

Comment:
before this change the non-qualified `else` would act as a catch-all logger for error which seemed useful but now that is gone. just checking that that's intended. if the `else`'s capture all the `download_result` cases then using a `switch` on the `download_result` strong type will let the compiler notify us if any cases aren't handled in the future.

Reply:
The control flow in this function is a bit hard to follow :) We essentially have two places where we switch on errors: inside and outside the loop. But probably fixing it is outside the scope of the pr.

Anyway, because in case of failed/not_found we should exit the retry loop, maybe we could move logging to the switch statement there and just have the timeout case logged in the if statement here.